docs/examples/irradiance-transposition/plot_rtranpose_year.py

```diff
@@ -0,0 +1,134 @@
+"""
+Reverse transposition using one year of hourly data
+===================================================
+
+With a brief look at accuracy and speed.
+
+Author: Anton Driesse
+
+"""
+
+# %%
+#
+# Introduction
+# ------------
+# In this example we start with a TMY file and calculate POA global irradiance.
+# Then, we use :py:func:`pvlib.irradiance.rtranspose_driesse_2023` to estimate
+# the original GHI from POA global.  Details of the method found in [1]_.
+#
+# References
+# ----------
+# .. [1] A. Driesse, A. Jensen, R. Perez, A Continuous Form of the Perez
+#     Diffuse Sky Model for Forward and Reverse Transposition, accepted
+#     for publication in the Solar Energy Journal.
+#
+
+import os
+import pandas as pd
+
+import matplotlib.pyplot as plt
+
+import pvlib
+from pvlib import iotools, location
+from pvlib.irradiance import (get_extra_radiation,
+                              get_total_irradiance,
+                              rtranspose_driesse_2023,
+                              aoi,
+                              )
+
+from timeit import timeit
+
+# %%
+#
+# Read a TMY3 file containing weather data and select needed columns.
+#
+
+PVLIB_DIR = pvlib.__path__[0]
+DATA_FILE = os.path.join(PVLIB_DIR, 'data', '723170TYA.CSV')
+
+tmy, metadata = iotools.read_tmy3(DATA_FILE, coerce_year=1990,
+                                  map_variables=True)
+
+df = pd.DataFrame({'ghi': tmy['ghi'], 'dhi': tmy['dhi'], 'dni': tmy['dni'],
+                   'temp_air': tmy['temp_air'],
+                   'wind_speed': tmy['wind_speed'],
+                   })
+
+# %%
+#
+# Shift the timestamps to the middle of the hour and calculate sun positions.
+#
+
+df.index = df.index - pd.Timedelta(minutes=30)
+
+loc = location.Location.from_tmy(metadata)
+solpos = loc.get_solarposition(df.index)
+
+# %%
+#
+# Estimate global irradiance on a fixed-tilt array (forward transposition).
+# The array is tilted 30 degrees and oriented 30 degrees east of south.
+#
+
+TILT = 30
+ORIENT = 150
+
+df['dni_extra'] = get_extra_radiation(df.index)
+
+total_irrad = get_total_irradiance(TILT, ORIENT,
+                                   solpos.apparent_zenith, solpos.azimuth,
+                                   df.dni, df.ghi, df.dhi,
+                                   dni_extra=df.dni_extra,
+                                   model='perez-driesse')
+
+df['poa_global'] = total_irrad.poa_global
+df['aoi'] = aoi(TILT, ORIENT, solpos.apparent_zenith, solpos.azimuth)
+
+# %%
+#
+# Now estimate ghi from poa_global using reverse transposition
+# The algorithm uses a simple bisection search.
+#
+
+df['ghi_rev'] = rtranspose_driesse_2023(TILT, ORIENT,
+                                        solpos.apparent_zenith, solpos.azimuth,
+                                        df.poa_global,
+                                        dni_extra=df.dni_extra)
+
+# %%
+#
+# Let's see how long this takes...
```

Comment:
It's a bit of a shame that this script takes over a minute to run in the CI, which makes building the gallery take 3-4 times as long as without it.  A not-very-invasive improvement would be to time the function the first time it runs so there is no need to call it again for the timing.  But it would still take >30s even with that.  Maybe the dataset be reduced in size without making the example useless?

Reply:
Yes, I don't know how to get the timing and output at the same time.  I could take a random sample of say 10% of the points.  I don't know how to get the results and the timing from timeit, but I could also just state what the result was in the text.